src/hooks/use-site-details.tsx

```diff
@@ -227,7 +224,7 @@ export function SiteDetailsProvider( { children }: SiteDetailsProviderProps ) {
 					customDomain,
 					enableHttps
 				);
-				const newSite = data.find( ( site ) => site.path === path );
+				const newSite = data.sites.find( ( site ) => site.path === path );
```

Comment:
Not 100% happy about the `createSite` function returning the entire user config object… It might be worthwhile to update `withUserDataWrite` function to also support returning values to the caller (the same way that `withAppdataWrite` works).